src/Mesh/Interpolation.jl

```diff
@@ -273,185 +550,686 @@ struct InterpolationCubedSphere{T <: Integer, FT <: AbstractFloat}
+    Jac31 = (C13*v[1] + C23*v[2] + C33*v[3]) / det
+
+    v[1] = Jac11; v[2] = Jac21; v[3] = Jac31
 
     return nothing 
 end
 #--------------------------------------------------------
-function interpolate_cubed_sphere!(intrp_cs::InterpolationCubedSphere, sv::AbstractArray{FT}, st_no::T) where {T <: Integer, FT <: AbstractFloat}
-    qm1 = intrp_cs.poly_order + 1
-    Nel = length(intrp_cs.realelems)
-    m1_r, m1_w = GaussQuadrature.legendre(FT,qm1,GaussQuadrature.both)
-    wb = Elements.baryweights(m1_r)
-    phir = MVector{qm1,FT}(undef)
-    phis = MVector{qm1,FT}(undef)
-    phit = MVector{qm1,FT}(undef)
-
-    vout    = FT(0)
-    vout_ii = FT(0)
-    vout_ij = FT(0)
-
-    for el in 1:Nel #-----for each element elno 
-        np = length(intrp_cs.ξ1[el])
-        lag    = @view sv[:,st_no,el]
-        for i in 1:np # interpolating point-by-point
-            interpolationvector_1pt!(m1_r,intrp_cs.ξ1[el][i],wb,phir)
-            interpolationvector_1pt!(m1_r,intrp_cs.ξ2[el][i],wb,phis)
-            interpolationvector_1pt!(m1_r,intrp_cs.ξ3[el][i],wb,phit)
-            vout = 0.0
-            for ik in 1:qm1
-                vout_ij = 0.0
-                for ij in 1:qm1 
-                    vout_ii = 0.0 
-                    for ii in 1:qm1
-                        @inbounds vout_ii += lag[ii + (ij-1)*qm1 + (ik-1)*qm1*qm1] * phir[ii]
+"""
+    interpolate_local!(intrp_cs::InterpolationCubedSphere{FT,T,FTV,FTVD,TVD,UI8AD,UI16V,UI16VD,I32V},
+                                               sv::AbstractArray{FT}, v::AbstractArray{FT}) where {FT <: AbstractFloat, 
+                                                                                                    T <: Int,
+                                                                                                  FTV <: AbstractVector{FT},
+                                                                                                 FTVD <: AbstractVector{FT},
+                                                                                                  TVD <: AbstractVector{T},
+                                                                                                UI8AD <: AbstractArray{UInt8,2},
+                                                                                                UI16V <: AbstractVector{UInt16},
+                                                                                               UI16VD <: AbstractVector{UInt16},
+                                                                                                 I32V <: AbstractVector{Int32}} 
+
+This interpolation function works for cubed spherical shell geometry.
+
+# input
+ - `intrp_cs` Initialized cubed sphere structure
+ - `sv` State Array consisting of various variables on the discontinuous Galerkin grid
+ - `v`  Interpolated variables
+"""
+function interpolate_local!(intrp_cs::InterpolationCubedSphere{FT,T,FTV,FTVD,TVD,UI8AD,UI16V,UI16VD,I32V},
```

Comment:
```suggestion
function interpolate_local!(intrp_cs::InterpolationCubedSphere{FT},
```
and remove unnecessary clauses and in docs